scripts/o3de/o3de/repo_properties.py

```diff
@@ -263,6 +264,32 @@ def _edit_objects(object_typename:str,
 
     repo_json[f'{object_typename}s_data'] = repo_objects_data
 
+def _auto_update_json(object_type: str or list,
+                      repo_path: pathlib.Path = None,
+                      repo_json: dict = None,
```

Comment:
These arguments should not have default values if they are required. I am assuming at least `repo_path` is required since the call `    repo_directory = os.path.dirname(repo_path)` below will raise a type error